packages/account-abstraction/src/SmartAccountAPI.ts

```diff
@@ -188,49 +190,40 @@ export class SmartAccountAPI extends BaseAccountAPI {
 
     const initCode = await this.getInitCode()
 
-    const verificationGasLimit = BigNumber.from(await this.getVerificationGasLimit())
-
-    let { maxFeePerGas, maxPriorityFeePerGas } = info
-    // Note: Custom should be equal if it's for non EIP1559
-
-    if (maxFeePerGas == null || maxPriorityFeePerGas == null) {
-      const feeData = await this.provider.getFeeData()
-      Logger.log('EIP1559 feeData', feeData)
-      const chainId = this.clientConfig.chainId
-      Logger.log('chainId is', chainId)
-      // Can do based on non EIP1559 chainId
-      if (EIP1559_UNSUPPORTED_NETWORKS.includes(chainId)) {
-        maxFeePerGas = feeData.gasPrice ?? (await this.provider.getGasPrice()) ?? undefined
-        maxPriorityFeePerGas = feeData.gasPrice ?? (await this.provider.getGasPrice()) ?? undefined
-      }
-      if (maxFeePerGas == null) {
-        maxFeePerGas = feeData.maxFeePerGas ?? undefined // ethers.BigNumber.from('100000000000')
-      }
-      if (maxPriorityFeePerGas == null) {
-        maxPriorityFeePerGas = feeData.maxPriorityFeePerGas ?? undefined // ethers.BigNumber.from('35000000000')
-      }
-    }
-    if (!maxFeePerGas || !maxPriorityFeePerGas) {
-      throw new Error('maxFeePerGas or maxPriorityFeePerGas values cannot be null')
-    }
-
-    Logger.log('fees being used: maxFeePerGas ', maxFeePerGas)
-    Logger.log('fees being used: maxPriorityFeePerGas ', maxPriorityFeePerGas)
     /* eslint-disable  @typescript-eslint/no-explicit-any */
     const partialUserOp: any = {
       sender: await this.getAccountAddress(),
       nonce: await this.nonce(),
       initCode,
       callData,
       callGasLimit,
-      verificationGasLimit,
-      maxFeePerGas,
-      maxPriorityFeePerGas
+      paymasterAndData: '0x'
+    }
+
+    let { maxFeePerGas, maxPriorityFeePerGas } = info
+    const chainId = this.clientConfig.chainId
+
+    const feeData = await this.httpRpcClient.getUserOpGasAndGasPrices(partialUserOp)
+    Logger.log('feeData', feeData)
+
+    if (EIP1559_UNSUPPORTED_NETWORKS.includes(chainId)) {
+      maxFeePerGas = undefined
```

Comment:
if we manage to get feeData from bundler assumption is it would have already considered EIP1559 support and returned valid and same values for maxFeePerGas and maxPriorityFeePerGas